src/Tokens/Models/UserIdToken/UserIdToken.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+namespace Pinnacle\OpenIdConnect\Tokens\Models\UserIdToken;
+
+use GuzzleHttp\Psr7\Uri;
+use Pinnacle\OpenIdConnect\Tokens\Exceptions\InvalidUserIdTokenException;
+use Pinnacle\OpenIdConnect\Tokens\Exceptions\MissingRequiredClaimKeyException;
+use Pinnacle\OpenIdConnect\Tokens\Exceptions\UserIdTokenHasExpiredException;
+use Pinnacle\OpenIdConnect\Tokens\Models\UserIdToken\Constants\ClaimKey;
+
+class UserIdToken
+{
+    /**
+     * @var array
+     */
+    private array             $parsedValues;
+
+    private Uri               $issuerIdentifier;
+
+    private SubjectIdentifier $subjectIdentifier;
+
+    private Audiences         $audiences;
+
+    private int               $expirationTime;
+
+    private int               $issuedTime;
+
+    /**
+     * @throws InvalidUserIdTokenException
+     * @throws MissingRequiredClaimKeyException
+     * @throws UserIdTokenHasExpiredException
+     */
+    public function __construct(private string $token)
+    {
+        if (trim($this->token) === '') {
+            throw new InvalidUserIdTokenException(
+                sprintf('%s was provided an empty string in the constructor.', self::class)
+            );
+        }
+
+        $this->parsedValues = self::parseToken();
+
+        $this->assertParsedValuesContainRequiredClaims();
+
+        $this->issuerIdentifier  = new Uri($this->findClaimByKey(ClaimKey::ISSUER_IDENTIFIER()->getValue()));
+        $this->subjectIdentifier = new SubjectIdentifier(
+            $this->findClaimByKey(ClaimKey::SUBJECT_IDENTIFIER()->getValue())
+        );
+        $this->audiences         = new Audiences($this->findClaimByKey(ClaimKey::AUDIENCES()->getValue()));
+        $this->expirationTime    = (int)$this->findClaimByKey(ClaimKey::EXPIRATION_TIME()->getValue());
+        $this->issuedTime        = (int)$this->findClaimByKey(ClaimKey::ISSUED_TIME()->getValue());
```

Comment:
Many of these can throw exceptions for invalid values. We may want to add a `try/catch` and convert to a library exception type.